### PR TITLE
Fix/analysis module feedback

### PR DESCRIPTION
**Pull request by AdityaKhatri**
- Addresses https://github.com/the-deep/client/issues/1703, https://github.com/the-deep/client/issues/1716, https://github.com/the-deep/client/issues/1718, https://github.com/the-deep/client/issues/1715,  https://github.com/the-deep/client/issues/1717, https://github.com/the-deep/client/issues/1720, https://github.com/the-deep/client/issues/1719, https://github.com/the-deep/client/issues/1715
- Depends on https://github.com/the-deep/server/pull/769

## Changes

* Fix multiple requests on the analysis module page
* Create overview and framework overview charts
* Add continue/view tagging button on home page
* Use the new form in filter entries form for pillar analysis except for date filter and time filter
* Fix minor feedback for analysis module and pillar analysis page
* Make `include in report` button functional
* Remove fake request for discarded entry tags list
* Create new excerpt output component except for dropped text visibility

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files, or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] permission checks
- [ ] translations

>NOTE: I apologize for the large PR, but all of the tasks overlapped with each other and it was difficult to separate out.

### CODE REVIEW

File: src/entities/entries.js [hunk 1]
```diff
@@ -89,6 +89,15 @@ export const processEntryFilters = (filters, framework, geoOptions, hideMatrixFi
...
             ]);
+        } else if (filterKey === 'verified') {
+            const isVerified = typeof filterOptions === 'string'
+                ? filterOptions === 'true'
+                : filterOptions;
+
+            result.push([
+                'verified',
+                isVerified,
+            ]);
```
Comment (samshara) [inline src/entities/entries.js]: why are we doing this?

File: src/newViews/AnalysisModule/Analysis/AnalysisPillar/index.tsx [hunk 2]
```diff
@@ -58,11 +61,20 @@ function AnalysisPillar(props: ComponentProps) {
 
     const disabled = pendingPillarDelete;
 
+    const onDeleteConfirmClick = useCallback(() => {
+        onDelete(pillarId);
+    }, [onDelete, pillarId]);
+
+    const onCloneConfirmClick = useCallback(() => {
+        console.warn('cloning', pillarId);
```
Comment (samshara) [inline src/newViews/AnalysisModule/Analysis/AnalysisPillar/index.tsx]: it's better to remove this.

File: src/newViews/AnalysisModule/Analysis/PillarAssignment/index.tsx [hunk 3]
```diff
@@ -1,23 +1,33 @@
...
+        entriesAnalyzed = 0,
+        totalEntries = 0,
     } = props;
 
+    let statusLabel = 'In progress';
+    if (entriesAnalyzed === totalEntries && entriesAnalyzed > 0) {
+        statusLabel = 'Analysis Completed';
+    } else if (totalEntries === 0) {
+        statusLabel = 'No Analysis';
+    }
```
Comment (samshara) [inline src/newViews/AnalysisModule/Analysis/PillarAssignment/index.tsx]: use of string translations is recommended

File: src/newViews/AnalysisModule/index.tsx [hunk 4]
```diff
@@ -98,8 +102,8 @@ interface AuthoringOrganizations {
 /*
 const dummyAuthoringOrganizations: AuthoringOrganizations[] = [
     { count: 10, organizationTypeId: 1, organizationTypeTitle: 'NGOs' },
-    { count: 6, organizationTypeId: 1, organizationTypeTitle: 'Government' },
-    { count: 16, organizationTypeId: 1, organizationTypeTitle: 'UN Agency' },
+    { count: 10, organizationTypeId: 1, organizationTypeTitle: 'Government' },
+    { count: 10, organizationTypeId: 1, organizationTypeTitle: 'UN Agency' },
 ];
  */
```
Comment (samshara) [inline src/newViews/AnalysisModule/index.tsx]: any reason we are keeping this

File: src/newViews/AnalysisModule/index.tsx [hunk 5]
```diff
@@ -327,7 +332,12 @@ function AnalysisModule(props: AnalysisModuleProps) {
                         icon={<IoCheckmarkCircle />}
                     />
                 </div>
-                <Card className={styles.pieChartContainer}>
+                <ContainerCard
+                    className={styles.pieChartContainer}
+                    footerIcons={(
+                        <p>Sources by type</p>
```
Comment (samshara) [inline src/newViews/AnalysisModule/index.tsx]: use of translations is recommended

File: src/newViews/PillarAnalysis/EntriesFilterForm/index.tsx [hunk 6]
```diff
@@ -72,6 +74,20 @@ const commentStatusOptions: KeyValueElement[] = [
...
 
+type FormType = FaramValues;
+
+type FormSchema = ObjectSchema<FormType>;
+type FormSchemaFields = ReturnType<FormSchema['fields']>;
+
+const schema: FormSchema = {
+    fields: (): FormSchemaFields => ({
+        created_by: [],
+    }),
```
Comment (samshara) [inline src/newViews/PillarAnalysis/EntriesFilterForm/index.tsx]: shouldn't we add other fields too?

File: src/newViews/PillarAnalysis/EntriesFilterForm/index.tsx [hunk 7]
```diff
@@ -164,111 +173,104 @@ function EntriesFilterForm(props: OwnProps) {
...
             )}
+            onSubmit={createSubmitHandler(validate, onErrorSet, handleSubmit)}
         >
             <MultiSelectInput
                 className={styles.filter}
-                faramElementName="created_by"
+                name="created_by"
                 keySelector={optionKeySelector}
                 labelSelector={optionLabelSelector}
+                value={value?.created_by as (string[] | undefined)}
```
Comment (samshara) [inline src/newViews/PillarAnalysis/EntriesFilterForm/index.tsx:206]: any reason we are using type assertions here? Shouldn't we define the types in FormType ?

File: src/newViews/PillarAnalysis/EntryItem/ExcerptOutput/index.tsx [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+                    value={props.dataSeries}
+                    onEntryStateChange={setEntryState}
+                    entryState={entryState}
+                />
+            );
+         */
+        return (
+            <p className={className}>
+                Quantitative data is not supported at the moment.
+            </p>
```
Comment (samshara) [inline src/newViews/PillarAnalysis/EntryItem/ExcerptOutput/index.tsx]: use of translations is recommended

**Review by samshara — CHANGES_REQUESTED** [review]

File: src/entities/entries.js (hunk 1 above)
Comment (AdityaKhatri) [reply src/entities/entries.js]: New SelectInput doesn't support boolean. This is a hack.

File: src/newViews/AnalysisModule/Analysis/AnalysisPillar/index.tsx (hunk 2 above)
Comment (AdityaKhatri) [reply src/newViews/AnalysisModule/Analysis/AnalysisPillar/index.tsx]: I'll add this request.

File: src/newViews/AnalysisModule/index.tsx (hunk 4 above)
Comment (AdityaKhatri) [reply src/newViews/AnalysisModule/index.tsx]: To test when we don't have data.
I'll remove this.

File: src/newViews/PillarAnalysis/EntriesFilterForm/index.tsx (hunk 6 above)
Comment (AdityaKhatri) [reply src/newViews/PillarAnalysis/EntriesFilterForm/index.tsx]: As all fields are dynamic, I am planning to remove this as well.

File: src/newViews/PillarAnalysis/EntriesFilterForm/index.tsx (hunk 7 above)
Comment (AdityaKhatri) [reply src/newViews/PillarAnalysis/EntriesFilterForm/index.tsx:206]: We don't really know types for all the auto-generated filter fields.
We'll need to change the data structure of widgets' filter in future to completely make this type safe.
For now, there is no other way.

**Review by samshara — APPROVED** [review]

File: src/typings/entry.ts [hunk 9]
```diff
@@ -138,6 +138,6 @@ export interface TocCountMap {
 }
 
 export interface EntryOptions {
-    createdBy: KeyValueElement;
+    createdBy: KeyValueElement[];
```
Comment (tnagorra) [inline src/typings/entry.ts:141]: Is this correct?

File: src/newViews/PillarAnalysis/index.tsx [hunk 10]
```diff
@@ -246,13 +238,33 @@ function PillarAnalysis(props: Props) {
                 o => o.id,
             );
             setFiltersValue(newFilters);
+            // eslint-disable-next-line max-len
+            let analyticalStatements: PartialAnalyticalStatementType[] = response.analyticalStatements ?? [];
+            if (
+                (response.analyticalStatements?.length ?? 0) === 0
+                || response.versionId === 1
+            ) {
```
Comment (tnagorra) [inline src/newViews/PillarAnalysis/index.tsx]: Let's comment why we are doing this

File: src/newViews/PillarAnalysis/EntriesFilterForm/index.tsx [hunk 11]
```diff
@@ -124,24 +133,24 @@ function EntriesFilterForm(props: OwnProps) {
...
+        onErrorSet,
+        value,
+        onValueSet,
+        onValueChange,
+    } = useForm(initialValue, schema);
+
+    useEffect(() => {
+        if (filtersValue) {
+            onValueSet(filtersValue);
+        }
```
Comment (tnagorra) [inline src/newViews/PillarAnalysis/EntriesFilterForm/index.tsx]: Why add this `if` check

File: src/components/input/GeoMultiSelectInput/index.tsx [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+>;
+
+function GeoMultiSelectInput<K extends string>(props: GeoSelectInputProps<K>) {
+    const {
+        className,
+        options,
+        ...otherProps
+    } = props;
+
+    const [, setOpened] = useState(false);
```
Comment (tnagorra) [inline src/components/input/GeoMultiSelectInput/index.tsx]: Do we need this?

File: src/newViews/AnalysisModule/Analysis/PillarList/index.tsx [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+            method: 'GET',
+            query: queryOptions,
+        },
+    );
+
+    // NOTE: Whenever the details of the analysis is changed, we refetch all the pillar
+    // analysis of that analysis
+    useEffect(() => {
+        pillarGetTrigger();
+    }, [pillarGetTrigger, modifiedAt]);
```
Comment (tnagorra) [inline src/newViews/AnalysisModule/Analysis/PillarList/index.tsx]: Why not get call this when we are specifically modifying the "modifiedAt"

File: src/newViews/PillarAnalysis/EntriesFilterForm/FrameworkFilter/index.tsx [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+    value: FaramValues;
+    onValueChange: (...entries: EntriesAsList<FaramValues>) => void;
+}
+
+function FrameworkFilter(props: Props) {
+    const {
+        title,
+        filterKey,
+        filter,
+        geoOptions = {},
```
Comment (tnagorra) [inline src/newViews/PillarAnalysis/EntriesFilterForm/FrameworkFilter/index.tsx]: Let's not create inline object here

File: src/newViews/PillarAnalysis/EntriesFilterForm/FrameworkFilter/index.tsx [hunk 15]
```diff
@@ -0,0 +1,123 @@
...
+        className,
+        value,
+        onValueChange,
+    } = props;
+
+    if (!filter || !filter.type) {
```
Comment (tnagorra) [inline src/newViews/PillarAnalysis/EntriesFilterForm/FrameworkFilter/index.tsx]:
```suggestion
    if (!filter?.type) {
```

File: src/components/input/GeoMultiSelectInput/index.tsx (hunk 12 above)
Comment (frozenhelium) [reply src/components/input/GeoMultiSelectInput/index.tsx]: Why is this here?

File: src/typings/entry.ts (hunk 9 above)
Comment (AdityaKhatri) [reply src/typings/entry.ts:141]: Yes. createdBy is a list of users

File: src/entities/entries.js [hunk 16]
```diff
@@ -89,6 +89,15 @@ export const processEntryFilters = (filters, framework, geoOptions, hideMatrixFi
                 `${filterKey}__lt`,
                 getDateWithTimezone(encodeDate(endDate)),
             ]);
+        } else if (filterKey === 'verified') {
+            const isVerified = typeof filterOptions === 'string'
+                ? filterOptions === 'true'
+                : filterOptions;
```
Comment (frozenhelium) [inline src/entities/entries.js]:
```suggestion
            const isVerified = String(filterOptions) === 'true';
```

File: src/newViews/AnalysisModule/Analysis/AnalysisPillar/index.tsx [hunk 17]
```diff
@@ -104,30 +129,42 @@ function AnalysisPillar(props: ComponentProps) {
...
+                                    {
+                                        entriesCount: statement.analyticalEntries.length,
+                                    },
+                                )
+                            }
                         />
                     ))}
                 />
-            </div>
+            )}
```
Comment (frozenhelium) [inline src/newViews/AnalysisModule/Analysis/AnalysisPillar/index.tsx]: Maybe use List

File: src/newViews/AnalysisModule/Analysis/PillarAssignment/index.tsx [hunk 18]
```diff
@@ -1,23 +1,35 @@
...
 
+    let statusLabel = _ts('analysis', 'inProgressTagLabel');
+    if (entriesAnalyzed === totalEntries && entriesAnalyzed > 0) {
+        statusLabel = _ts('analysis', 'analysisCompletedTagLabel');
+    } else if (totalEntries === 0) {
+        statusLabel = _ts('analysis', 'noAnalysisTagLabel');
+    }
+    const isAnalysisCompleted = entriesAnalyzed === totalEntries && totalEntries > 0;
+
     return (
```
Comment (frozenhelium) [inline src/newViews/AnalysisModule/Analysis/PillarAssignment/index.tsx:33]: If possible, lets use `TextOutput` for this

File: src/newViews/AnalysisModule/Analysis/PillarList/index.tsx [hunk 19]
```diff
@@ -0,0 +1,128 @@
...
+    MultiResponse,
+    AnalysisPillars,
+} from '#typings';
+
+import AnalysisPillar from '../AnalysisPillar';
+
+import styles from './styles.scss';
+
+const MAX_ITEMS_PER_PAGE = 5;
+const keySelector = (item: AnalysisPillars) => (item.id);
```
Comment (frozenhelium) [inline src/newViews/AnalysisModule/Analysis/PillarList/index.tsx]:
```suggestion
const keySelector = (item: AnalysisPillars) => item.id;
```

File: src/newViews/AnalysisModule/Analysis/PillarList/index.tsx [hunk 20]
```diff
@@ -0,0 +1,128 @@
...
+    const {
+        className,
+        createdAt,
+        modifiedAt,
+        activeProject,
+        onAnalysisPillarDelete,
+        analysisId,
+    } = props;
+
+    const [activePage, setActivePage] = useState<number>(1);
```
Comment (frozenhelium) [inline src/newViews/AnalysisModule/Analysis/PillarList/index.tsx]:
```suggestion
    const [activePage, setActivePage] = useState(1);
```
The typing will be automatically inferred

File: src/newViews/AnalysisModule/Analysis/PillarList/styles.scss [hunk 21]
```diff
@@ -0,0 +1,15 @@
+.container {
+    background-color: var(--dui-color-foreground);
+    padding: 0;
+
+    .content {
+        padding: 0;
```
Comment (frozenhelium) [inline src/newViews/AnalysisModule/Analysis/PillarList/styles.scss:6]: Hmm, do we need to set the `padding: 0`?

File: src/newViews/AnalysisModule/Analysis/index.tsx [hunk 22]
```diff
@@ -1,34 +1,70 @@
...
+    if (value === 0) {
+        return null;
+    }
+
+    return (
+        <g>
+            <text
+                x={x + (width / 2)}
+                y={y - radius}
+                fill="#717171"
```
Comment (frozenhelium) [inline src/newViews/AnalysisModule/Analysis/index.tsx]: We should move this to styling (with the help of className)

File: src/newViews/AnalysisModule/Analysis/index.tsx [hunk 23]
```diff
@@ -171,30 +166,43 @@ function Analysis(props: ComponentProps) {
...
-                    <QuickActionButton
+                    <Button
                         name="edit"
                         onClick={handleEditClick}
                         disabled={disabled}
+                        variant="tertiary"
+                        icons={(
+                            <FiEdit2 />
+                        )}
                     >
```
Comment (frozenhelium) [inline src/newViews/AnalysisModule/Analysis/index.tsx]:
```suggestion
                    <Button
                        name={analysisId}
                        onClick={onEdit}
                        disabled={disabled}
                        variant="tertiary"
                        icons={(
                            <FiEdit2 />
                        )}
                    >
```

File: src/newViews/AnalysisModule/Analysis/index.tsx [hunk 24]
```diff
@@ -218,39 +226,94 @@ function Analysis(props: ComponentProps) {
...
+                >
+                    <ResponsiveContainer className={styles.responsiveContainer}>
+                        <BarChart
+                            data={barChartData}
+                            margin={{
+                                top: 5,
+                                right: 30,
+                                left: 20,
+                                bottom: 5,
+                            }}
```
Comment (frozenhelium) [inline src/newViews/AnalysisModule/Analysis/index.tsx]: Lets define this as a constant

File: src/newViews/AnalysisModule/Analysis/index.tsx [hunk 25]
```diff
@@ -218,39 +226,94 @@ function Analysis(props: ComponentProps) {
...
+                                tickLine={false}
+                            />
+                            <CartesianGrid
+                                vertical={false}
+                                stroke="var(--dui-color-background-information)"
+                            />
+                            <YAxis
+                                axisLine={false}
+                                domain={[0, 100]}
+                                tickCount={5}
```
Comment (frozenhelium) [inline src/newViews/AnalysisModule/Analysis/index.tsx]: Lets define `tickCount` as a counstant

File: src/newViews/AnalysisModule/Analysis/index.tsx [hunk 26]
```diff
@@ -218,39 +226,94 @@ function Analysis(props: ComponentProps) {
...
+                                tickLine={false}
+                            />
+                            <Bar
+                                dataKey="percent"
+                                fill="var(--dui-color-accent)"
+                                background={{
+                                    fill: 'var(--dui-color-background-information)',
+                                }}
+                                maxBarSize={16}
```
Comment (frozenhelium) [inline src/newViews/AnalysisModule/Analysis/index.tsx]: Lets define it as a constant

File: src/newViews/AnalysisModule/Analysis/styles.scss [hunk 27]
```diff
@@ -5,72 +5,101 @@
     border: var(--width-separator-thin) solid var(--dui-border-color-card);
     border-radius: var(--dui-border-radius-card);
     background-color: var(--color-foreground);
-    overflow: auto;
+    // NOTE: I don't want to do this
```
Comment (frozenhelium) [inline src/newViews/AnalysisModule/Analysis/styles.scss:8]:
```suggestion
    // NOTE: We don't want to do this
```

File: src/newViews/AnalysisModule/AnalysisEditModal/index.tsx [hunk 28]
```diff
@@ -113,31 +114,38 @@ function AnalysisEditModal(props: AnalysisEditModalProps) {
...
+        skip: !analysisToEdit,
+        url: `server://projects/${projectId}/analysis/${analysisToEdit}/`,
+        onSuccess: (response) => {
+            onValueSet({
+                teamLead: response.teamLead,
+                title: response.title,
+                analysisPillar: response.analysisPillar.map(ap => ({
+                    key: String(ap.id),
+                    assignee: ap.assignee,
+                    filters: ap.filters ? ap.filters.map(f => f.uniqueId) : undefined,
```
Comment (frozenhelium) [inline src/newViews/AnalysisModule/AnalysisEditModal/index.tsx]:
```suggestion
                    filters: ap.filters?.map(f => f.uniqueId),
```

File: src/newViews/AnalysisModule/index.tsx [hunk 29]
```diff
@@ -327,7 +324,12 @@ function AnalysisModule(props: AnalysisModuleProps) {
                         icon={<IoCheckmarkCircle />}
                     />
                 </div>
-                <Card className={styles.pieChartContainer}>
+                <ContainerCard
+                    className={styles.pieChartContainer}
+                    footerIcons={(
+                        <p>{_ts('analysis', 'sourcesByTypeLabel')}</p>
+                    )}
```
Comment (frozenhelium) [inline src/newViews/AnalysisModule/index.tsx]:
```suggestion
                    footerContent={_ts('analysis', 'sourcesByTypeLabel')}
```

File: src/newViews/PillarAnalysis/AnalyticalStatementInput/styles.scss [hunk 30]
```diff
@@ -22,6 +22,11 @@
             border-bottom: none;
             background-color: var(--dui-color-surface-informational);
             padding: calc(var(--dui-spacing-large) - var(--dui-spacing-medium));
+
+            .include-in-report-button {
+                padding: 0;
+                font-size: 1.3em;
```
Comment (frozenhelium) [inline src/newViews/PillarAnalysis/AnalyticalStatementInput/styles.scss]: Maybe use `big` prop in button

File: src/newViews/PillarAnalysis/AnalyticalStatementInput/styles.scss [hunk 31]
```diff
@@ -22,6 +22,11 @@
             border-bottom: none;
             background-color: var(--dui-color-surface-informational);
             padding: calc(var(--dui-spacing-large) - var(--dui-spacing-medium));
+
+            .include-in-report-button {
+                padding: 0;
```
Comment (frozenhelium) [inline src/newViews/PillarAnalysis/AnalyticalStatementInput/styles.scss]: We should not have to do this

File: src/newViews/PillarAnalysis/EntryItem/ExcerptOutput/index.tsx [hunk 32]
```diff
@@ -0,0 +1,72 @@
...
+        /*
+            return (
+                <DataSeries
+                    className={styles.dataSeries}
+                    value={props.dataSeries}
+                    onEntryStateChange={setEntryState}
+                    entryState={entryState}
+                />
+            );
+         */
```
Comment (frozenhelium) [inline src/newViews/PillarAnalysis/EntryItem/ExcerptOutput/index.tsx]: Lets remove this comment

File: src/newViews/PillarAnalysis/styles.scss [hunk 33]
```diff
@@ -68,7 +69,25 @@
...
+                    .tab-list-heading {
+                        display: flex;
+                        padding: 0;
+
+                        .tab-list {
+                            flex-grow: 1;
+
+                            >* {
+                                flex-grow: 1;
+                            }
```
Comment (frozenhelium) [inline src/newViews/PillarAnalysis/styles.scss]: Lets ensure the `display: flex` before doing `flex-grow: 1` in children

**Review by frozenhelium — CHANGES_REQUESTED** [review]

**Comment by frozenhelium** [comment]
Also, please, remove empty files.

File: src/newViews/PillarAnalysis/EntriesFilterForm/index.tsx (hunk 11 above)
Comment (AdityaKhatri) [reply src/newViews/PillarAnalysis/EntriesFilterForm/index.tsx]: we shouldn't set the form value as undefined which might be the case

File: src/newViews/AnalysisModule/Analysis/PillarList/index.tsx (hunk 13 above)
Comment (AdityaKhatri) [reply src/newViews/AnalysisModule/Analysis/PillarList/index.tsx]: The action that changes modifiedAt is at parent, that is why I've checked for change ni modifiedAt here.
Let me know if there is a better way to do this.

File: src/newViews/AnalysisModule/Analysis/PillarAssignment/index.tsx (hunk 18 above)
Comment (AdityaKhatri) [reply src/newViews/AnalysisModule/Analysis/PillarAssignment/index.tsx:33]: We are using Tag for this.

File: src/newViews/AnalysisModule/Analysis/PillarList/styles.scss (hunk 21 above)
Comment (AdityaKhatri) [reply src/newViews/AnalysisModule/Analysis/PillarList/styles.scss:6]: Yes, to comply with the design this was the best possible way. its adding unnecessary padding on top and bottom which made the UI a bit weird in this case.
Let me know if there is a better way. We can discuss it on a call.

File: src/newViews/AnalysisModule/Analysis/AnalysisPillar/index.tsx (hunk 17 above)
Comment (AdityaKhatri) [reply src/newViews/AnalysisModule/Analysis/AnalysisPillar/index.tsx]: It wasn't me :'(

File: src/typings/entry.ts (hunk 9 above)
Comment (tnagorra) [reply src/typings/entry.ts:141]: How can that be?

File: src/newViews/PillarAnalysis/EntriesFilterForm/index.tsx (hunk 11 above)
Comment (tnagorra) [reply src/newViews/PillarAnalysis/EntriesFilterForm/index.tsx]: why not?

File: src/newViews/AnalysisModule/Analysis/PillarList/index.tsx (hunk 13 above)
Comment (tnagorra) [reply src/newViews/AnalysisModule/Analysis/PillarList/index.tsx]: Parent passess a function as prop and calls it.

File: src/newViews/PillarAnalysis/index.tsx [hunk 34]
```diff
@@ -246,13 +238,36 @@ function PillarAnalysis(props: Props) {
                 o => o.id,
             );
             setFiltersValue(newFilters);
+            // eslint-disable-next-line max-len
+            let analyticalStatements: PartialAnalyticalStatementType[] = response.analyticalStatements ?? [];
+            if (
+                (response.analyticalStatements?.length ?? 0) === 0
+                || response.versionId === 1
```
Comment (tnagorra) [inline src/newViews/PillarAnalysis/index.tsx]: Shoudn't this be an `and` expression?
Also, check if versionId starts with 0

File: src/components/input/GeoMultiSelectInput/index.tsx [hunk 35]
```diff
@@ -0,0 +1,69 @@
...
+    const {
+        className,
+        options,
+        ...otherProps
+    } = props;
+
+    const [searchText, setSearchText] = useState<string>('');
+    const debouncedSearchText = useDebouncedValue(searchText);
+
+    const searchOptions = useMemo(() => {
```
Comment (tnagorra) [inline src/components/input/GeoMultiSelectInput/index.tsx]: Why not check for falsy debouncedSearchText before doing the filter?

File: src/components/input/GeoMultiSelectInput/index.tsx [hunk 36]
```diff
@@ -0,0 +1,69 @@
...
+        const filteredOptions = options?.filter((option) => {
+            const searchFilter = isFalsyString(debouncedSearchText)
+                || caseInsensitiveSubmatch(option.label, debouncedSearchText);
+            return searchFilter;
+        });
+        if (filteredOptions && (filteredOptions.length ?? 0 > 50)) {
+            filteredOptions.length = 50;
+        }
```
Comment (tnagorra) [inline src/components/input/GeoMultiSelectInput/index.tsx]: Let's just use slice / splice

File: src/newViews/PillarAnalysis/AnalyticalStatementInput/index.tsx [hunk 37]
```diff
@@ -168,6 +168,10 @@ function AnalyticalStatementInput(props: AnalyticalStatementInputProps) {
         [handleAnalyticalEntryDrop],
     );
 
+    const handleIncludeInReportChange = useCallback(() => {
+        onFieldChange((oldVal?: boolean) => !oldVal, 'includeInReport' as const);
+    }, [onFieldChange]);
+
```
Comment (tnagorra) [inline src/newViews/PillarAnalysis/AnalyticalStatementInput/index.tsx:174]: Why do we need this? The input should have done this by itself

Comment (AdityaKhatri) [reply src/newViews/PillarAnalysis/AnalyticalStatementInput/index.tsx:174]: Its a button. Should I change the name?

File: src/typings/entry.ts (hunk 9 above)
Comment (AdityaKhatri) [reply src/typings/entry.ts:141]: options of entries filter

File: src/newViews/AnalysisModule/Analysis/PillarList/index.tsx (hunk 13 above)
Comment (AdityaKhatri) [reply src/newViews/AnalysisModule/Analysis/PillarList/index.tsx]: the action resides on child and not parent.

File: src/newViews/PillarAnalysis/index.tsx [hunk 38]
```diff
@@ -76,33 +76,28 @@ export interface DiscardedTags {
...
-    },
-];
-
-// This is an aribitrary number
+// This is an aribtrary number
 const STATEMENTS_LIMIT = 30;
 
+const analysisEntriesRequestQuery = {
+    // NOTE: 30 columns x 50 rows
+    limit: 30 * 50,
```
Comment (tnagorra) [inline src/newViews/PillarAnalysis/index.tsx]: Why not mulitply using the defined constants?

File: src/newViews/AnalysisModule/Analysis/PillarList/index.tsx (hunk 13 above)
Comment (tnagorra) [reply src/newViews/AnalysisModule/Analysis/PillarList/index.tsx]: oh no!

File: src/newViews/PillarAnalysis/index.tsx (hunk 34 above)
Comment (tnagorra) [reply src/newViews/PillarAnalysis/index.tsx]: Did we check?

File: src/newViews/PillarAnalysis/AnalyticalStatementInput/index.tsx (hunk 37 above)
Comment (tnagorra) [reply src/newViews/PillarAnalysis/AnalyticalStatementInput/index.tsx:174]: Should have used a checkbox component instead.

**Pull request merged**